command/server_dev.go

```diff
@@ -0,0 +1,900 @@
+	}
+	resp, err = core.HandleRequest(ctx, r)
+	if err != nil {
+		return fmt.Errorf("error reading mount accessor: %w", err)
+	}
+	if resp.IsError() {
+		return fmt.Errorf("failed to read mount accessor: %w", resp.Error())
+	}
+	up, ok := resp.Data["userpass/"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("failed to read mount accessor")
+	}
+	accessor, ok := up["accessor"].(string)
+	if !ok {
+		return fmt.Errorf("failed to read mount accessor")
+	}
+
+	r = &logical.Request{
+		Operation:   logical.CreateOperation,
+		ClientToken: init.RootToken,
+		Path:        "identity/entity-alias",
+		Data: map[string]interface{}{
+			"name":           options.authUserpass.username,
+			"canonical_id":   id,
+			"mount_accessor": accessor,
+			"custom-metadata": map[string]string{
+				"account": "demo",
+			},
+		},
+	}
+	resp, err = core.HandleRequest(ctx, r)
+	if err != nil {
+		return fmt.Errorf("error creating entity: %w", err)
+	}
+	if resp.IsError() {
+		return fmt.Errorf("failed to create entity: %w", resp.Error())
+	}
+
+	return nil
+}
+
+func qsEnableAppRole(ctx context.Context, init *vault.InitResult, core *vault.Core, options *quickstartOptions) error {
+	r := &logical.Request{
+		Operation:   logical.UpdateOperation,
+		ClientToken: init.RootToken,
+		Path:        "sys/auth/approle",
+		Data: map[string]interface{}{
+			"type":        "approle",
+			"path":        "approle/",
+			"description": "sample approle method",
+		},
+	}
+	resp, err := core.HandleRequest(ctx, r)
+	if err != nil {
+		return fmt.Errorf("error creating approle: %w", err)
+	}
+	if resp.IsError() {
+		return fmt.Errorf("failed to create approle: %w", resp.Error())
+	}
+
+	r = &logical.Request{
+		Operation:   logical.UpdateOperation,
+		ClientToken: init.RootToken,
+		Path:        fmt.Sprintf("auth/approle/role/%s", options.authAppRole.roleName),
+		Data: map[string]interface{}{
+			"role_name":      options.authAppRole.roleName,
+			"token_policies": options.authAppRole.policies,
+		},
+	}
+	resp, err = core.HandleRequest(ctx, r)
+	if err != nil {
+		return fmt.Errorf("error creating approle role: %w", err)
+	}
+	if resp.IsError() {
+		return fmt.Errorf("failed to create approle role: %w", resp.Error())
+	}
+
+	r = &logical.Request{
+		Operation:   logical.ReadOperation,
+		ClientToken: init.RootToken,
+		Path:        fmt.Sprintf("auth/approle/role/%s/role-id", options.authAppRole.roleName),
+	}
+	resp, err = core.HandleRequest(ctx, r)
+	if err != nil {
+		return fmt.Errorf("error retrieving approle id: %w", err)
+	}
+	if resp.IsError() {
+		return fmt.Errorf("failed to retrieve approle id: %w", resp.Error())
+	}
+
+	roleID, ok := resp.Data["role_id"].(string)
+	if !ok {
+		return fmt.Errorf("failed to parse role id")
+	}
+	options.authAppRole.roleID = roleID
+
+	r = &logical.Request{
+		Operation:   logical.UpdateOperation,
+		ClientToken: init.RootToken,
+		Path:        fmt.Sprintf("auth/approle/role/%s/secret-id", options.authAppRole.roleName),
```

Comment:
How long does this secret-id last? And is it one-time use?

Reply:
With no options being provided I believe the secret id lasts forever and has infinite uses.
`vault read auth/approle/role/app-read-only`
```text                                                                                                                                                                                                                    
Key                        Value
---                        -----
bind_secret_id             true
local_secret_ids           false
secret_id_bound_cidrs      <nil>
secret_id_num_uses         0
secret_id_ttl              0s
token_bound_cidrs          []
token_explicit_max_ttl     0s
token_max_ttl              0s
token_no_default_policy    false
token_num_uses             0
token_period               0s
token_policies             [read-only]
token_ttl                  0s
token_type                 default
```